tensorflow_datasets/core/load.py

```diff
@@ -79,16 +79,19 @@ def builder_cls(name: str) -> Type[dataset_builder.DatasetBuilder]:
     DatasetNotFoundError: if `name` is unrecognized.
   """
   ds_name, kwargs = naming.parse_builder_name_kwargs(name)
-  #Checking whether the same dataset already exists in TFDS
-  dataset_common = ds_name.name in list_builders(with_community_datasets = False)
+  #Checking whether the dataset is not implemented
+  dataset_not_imp = (
+    ds_name.namespace == 'huggingface' and
+    ds_name.name in ['mnist', 'fashion_mnist']
```

Comment:
Let's not start to hardcode dataset name. We want this code to be as generic as possible. Also as explained in the other thread, silently overwriting the user choice feels confusing.

Instead, we should raise an error saying Array2d are not supported. It would be raise when user load('huggingface:mnist'),